sumologic-app-utils/src/sumoresource.py

```diff
@@ -1429,28 +1429,33 @@ def _get_root_folder_id(self):
         response = self.sumologic_cli.get_root_folder()
         return response["id"]
 
-    def import_monitor(self, folder_name, monitors3url, variables, suffix_date_time):
+    def import_monitor(self, folder_name, orgID, monitors3url, variables, suffix_date_time):
         date_format = "%d-%b-%Y %H:%M:%S"
         root_folder_id = self._get_root_folder_id()
         content = self._get_content_from_s3(monitors3url, variables)
         content["name"] = folder_name + " " + datetime.utcnow().strftime(date_format) if suffix_date_time \
             else folder_name
         response = self.sumologic_cli.import_monitors(root_folder_id, content)
         import_id = response["id"]
+        monitor_permission_payload = {"permissionStatementDefinitions": [{"permissions": ["Create","Read","Update","Delete","Manage"],"subjectType": "org","subjectId": orgID,"targetId": import_id}]}
+        self.sumologic_cli.set_monitors_permissions(monitor_permission_payload)
         print("ALERTS MONITORS - creation successful with ID %s and Name %s." % (import_id, folder_name))
         return {"ALERTS MONITORS": response["name"]}, import_id
 
-    def create(self, folder_name, monitors3url, variables, suffix_date_time=False, *args, **kwargs):
-        return self.import_monitor(folder_name, monitors3url, variables, suffix_date_time)
+    def create(self, folder_name, orgID, monitors3url, variables, suffix_date_time=False, *args, **kwargs):
+        return self.import_monitor(folder_name, orgID, monitors3url, variables, suffix_date_time)
 
-    def update(self, folder_id, folder_name, monitors3url, variables, suffix_date_time=False, retain_old_alerts=False, *args, **kwargs):
-        data, new_folder_id = self.create(folder_name, monitors3url, variables, suffix_date_time)
+    def update(self, folder_id, folder_name, orgID, monitors3url, variables, suffix_date_time=False, retain_old_alerts=False, *args, **kwargs):
+        data, new_folder_id = self.create(folder_name, orgID, monitors3url, variables, suffix_date_time)
         if retain_old_alerts:
             # Retaining old folder in the new folder as backup.
             try:
                 old_folder = self.sumologic_cli.export_monitors(folder_id)
                 old_folder["name"] = "Back Up " + old_folder["name"]
-                self.sumologic_cli.import_monitors(new_folder_id, old_folder)
+                response = self.sumologic_cli.import_monitors(new_folder_id, old_folder)
+                import_id = response["id"]
+                monitor_permission_payload = {"permissionStatementDefinitions": [{"permissions": ["Create","Read","Update","Delete","Manage"],"subjectType": "org","subjectId": orgID,"targetId": import_id}]}
```

Comment:
Backing up inside new folder. new folder is parent and is already given permission with create. Any subfolder will have the parents permission. Do we still need to provide permission to subfolder?

Reply:
Noted, New parent already has FGP so removing permissions api call.